packages/api/src/providers/fitbit.ts

```diff
@@ -291,17 +291,20 @@ export class Fitbit extends Provider implements OAuth2 {
 
   async fetchUserProfile(
     accessToken: string,
+    hasScope: boolean,
     extraHeaders?: { [k: string]: string }
-  ): Promise<FitbitUser> {
-    return this.oauth.fetchProviderData<FitbitUser>(
-      `${Fitbit.URL}/${Fitbit.API_PATH}/profile.json`,
-      accessToken,
-      async resp => {
-        return fitbitUserResp.parse(resp.data);
-      },
-      undefined,
-      extraHeaders
-    );
+  ): Promise<FitbitUser | undefined> {
+    if (hasScope) {
+      return this.oauth.fetchProviderData<FitbitUser>(
+        `${Fitbit.URL}/${Fitbit.API_PATH}/profile.json`,
+        accessToken,
+        async resp => {
+          return fitbitUserResp.parse(resp.data);
+        },
+        undefined,
+        extraHeaders
+      );
+    }
```

Comment:
Its odd to pass a boolean to a function that determines whether the function code should execute or not - wouldn't it be better to not call the function in the first place?